pyresample/test/test_bucket.py

```diff
@@ -17,328 +17,377 @@
-        # Scalar, integer resolution
-        self.assertEqual(bucket.round_to_resolution(5.5, 2.), 6)
-        # Scalar, non-integer resolution
-        self.assertEqual(bucket.round_to_resolution(5.5, 1.7), 5.1)
-        # List
-        self.assertTrue(np.all(bucket.round_to_resolution([4.2, 5.6], 2) ==
-                               np.array([4., 6.])))
-        # Numpy array
-        self.assertTrue(np.all(bucket.round_to_resolution(np.array([4.2, 5.6]), 2) ==
-                               np.array([4., 6.])))
-        # Dask array
-        self.assertTrue(
-            np.all(bucket.round_to_resolution(da.array([4.2, 5.6]), 2) ==
-                   np.array([4., 6.])))
-
-    def test_get_proj_coordinates(self):
-        """Test calculation of projection coordinates."""
-        prj = MagicMock()
-        prj.return_value = ([3.1, 3.1, 3.1], [4.8, 4.8, 4.8])
-        lons = [1., 1., 1.]
-        lats = [2., 2., 2.]
-        self.resampler.prj = prj
-        result = self.resampler._get_proj_coordinates(lons, lats)
-        prj.assert_called_once_with(lons, lats)
-        self.assertTrue(isinstance(result, np.ndarray))
-        np.testing.assert_equal(result, np.array([[3.1, 3.1, 3.1],
-                                                  [4.8, 4.8, 4.8]]))
-
-    def test_get_bucket_indices(self):
-        """Test calculation of array indices."""
-        # Ensure nothing is calculated
-        with dask.config.set(scheduler=CustomScheduler(max_computes=0)):
-            self.resampler._get_indices()
-        x_idxs, y_idxs = da.compute(self.resampler.x_idxs,
-                                    self.resampler.y_idxs)
-        np.testing.assert_equal(x_idxs, np.array([1710, 1710, 1707, 1705]))
-        np.testing.assert_equal(y_idxs, np.array([465, 465, 459, 455]))
-
-        # Additional small test case
-        adef = create_area_def(
-            area_id='test',
-            projection={'proj': 'latlong'},
-            width=2, height=2,
-            center=(0, 0),
-            resolution=10)
-        lons = da.from_array(
-            np.array([-10.0, -9.9, -0.1, 0, 0.1, 9.9, 10.0, -10.1, 0]),
-            chunks=2)
-        lats = da.from_array(
-            np.array([-10.0, -9.9, -0.1, 0, 0.1, 9.9, 10.0, 0, 10.1]),
-            chunks=2)
-        resampler = bucket.BucketResampler(source_lats=lats,
-                                           source_lons=lons,
-                                           target_area=adef)
+
+
+@pytest.fixture(scope="module")
+def lons():
+    """Get longitudes for tests."""
+    return da.from_array(np.array([[25., 25.], [25., 25.]]), chunks=CHUNKS)
+
+
+@pytest.fixture(scope="module")
+def lats():
+    """Get latitudes for tests."""
+    return da.from_array(np.array([[60., 60.00001], [60.2, 60.3]]), chunks=CHUNKS)
+
+
+@pytest.fixture(scope="module")
+def resampler(adef, lons, lats):
+    """Get initialised resampler for tests."""
+    return bucket.BucketResampler(adef, lons, lats)
```

Comment:
Is it a problem to make this module scoped when it seems the resampler object caches stuff in the instance variables:

```python
        self.x_idxs = None
        self.y_idxs = None
        self.y_idxs = None
        self.idxs = None
        self.idxs = None
        self._get_indices()
        self._get_indices()
        self.counts = None
        self.counts = None
```

Reply:
The cached variables are related to the indices computed from the input lat, lon and target areadef, which are also module-scoped fixtures. So in this setting I think this is fine. As a cross-check, I just switched to function-scoped and all tests still pass, so there is no cross-contamination. Also, the tests that need different inputs or patched/mocked components do not use the resampler fixture.